src/main/java/net/snowflake/client/jdbc/SnowflakeResultSetSerializableV1.java

```diff
@@ -903,18 +903,52 @@ private void setupTransientFields() throws SQLException {
    * ResultSet is sessionless.
    *
    * @return a ResultSet which represents for the data wrapped in the object
+   * @deprecated Please use new interface function getResultSet(ResultSetRetrieveConfig)
    */
   public ResultSet getResultSet() throws SQLException {
-    return getResultSet(null);
+    return getResultSetInternal(null);
+  }
+
+  /**
+   * Get ResultSet from the ResultSet Serializable object so that the user can access the data.
+   *
+   * @param resultSetRetrieveConfig The extra info to retrieve the result set.
+   * @return a ResultSet which represents for the data wrapped in the object
+   */
+  public ResultSet getResultSet(ResultSetRetrieveConfig resultSetRetrieveConfig)
+      throws SQLException {
+    // Adjust OCSP cache server if necessary.
+    try {
```

Comment:
To reviewers:
Adjust OCSP cache server if necessary.

Reply:
Are these APIs only called by Spark connector? Does the JDBC driver itself needs to be updated to use these APIs instead of the older deprecated ones?

Reply:
@sfc-gh-hchaturvedi
From snowflake, SC is the only user to use this API. But, both SC and JDBC are open source, it is difficult to say whether some customers use it.
We have unit test: SnowflakeResultSetSerializableIT to test this API. I have changed one occurrence to call getResultSet(ResultSetRetrieveConfig). I intend to leave two occurrences to call getResultSet(Properties) and getResultSet() for better JDBC test coverage.
If you think it is not necessary to call the 2 deprecated functions. please let me know. I will change them to call getResultSet(ResultSetRetrieveConfig).

Reply:
Latest status:
getResultSet() is removed.
getResultSet(Properties) is kept for compatibility.
All unit tests are updated to use getResultSet(ResultSetRetrieveConfig).

Reply:
Good choice to not break compatibility .